sklearn/ensemble/tests/test_forest.py

```diff
@@ -1330,3 +1330,96 @@ def test_forest_degenerate_feature_importances():
     gbr = RandomForestRegressor(n_estimators=10).fit(X, y)
     assert_array_equal(gbr.feature_importances_,
                        np.zeros(10, dtype=np.float64))
+
+
+@pytest.mark.parametrize('name', FOREST_CLASSIFIERS_REGRESSORS)
+@pytest.mark.parametrize(
+    'max_samples, exc_type, exc_msg',
+    [(int(1e9), ValueError,
+      "`max_samples` must be in range 1 to 6 but got value 1000000000"),
+     (1.0, ValueError,
+      "`max_samples` must be in range (0, 1) but got value 1.0"),
+     (2.0, ValueError,
+      "`max_samples` must be in range (0, 1) but got value 2.0"),
+     (0.0, ValueError,
+      "`max_samples` must be in range (0, 1) but got value 0.0"),
+     (np.nan, ValueError,
+      "`max_samples` must be in range (0, 1) but got value nan"),
+     (np.inf, ValueError,
+      "`max_samples` must be in range (0, 1) but got value inf"),
+     ('str max_samples?!', TypeError,
+      "`max_samples` should be int or float, but got type '<class \'str\'>'"),
+     (np.ones(2), TypeError,
+      "`max_samples` should be int or float, but got type "
+      "'<class \'numpy.ndarray\'>'")]
+)
+def test_max_samples_exceptions(name, max_samples, exc_type, exc_msg):
+    # Check invalid `max_samples` values
+    est = FOREST_CLASSIFIERS_REGRESSORS[name](max_samples=max_samples)
+    with pytest.raises(exc_type) as exc_info:
```

Comment:
matching partial message will be enough in fact

Reply:
I'm not sure what you mean by matching partial message?

Reply:
`pytest.raises(exc_type, match=exc_msg)` should work??